src/Shared/FileMatcher.cs

```diff
@@ -2334,7 +2334,8 @@ static string[] CreateArrayWithSingleItemIfNotExcluded(string filespecUnescaped,
                         else
                         {
                             //  Exclude base directory length is less than include base directory length.
-                            if (!state.BaseDirectory.StartsWith(excludeState.BaseDirectory, StringComparison.OrdinalIgnoreCase))
+                            if (!(state.BaseDirectory.StartsWith(excludeState.BaseDirectory, StringComparison.OrdinalIgnoreCase) &&
+                                  directorySeparatorCharacters.Contains(state.BaseDirectory[excludeState.BaseDirectory.Length])))
```

Comment:
Doesn't the previous `StartsWith` case also suffer from this?

Reply:
This one?
https://github.com/Microsoft/msbuild/blob/a4b44e00a085f90a279c0b454e2b1a8d8c77bea7/src/Shared/FileMatcher.cs#L2308-L2312

I had convinced myself that it didn't, but I can't remember why.

Reply:
I think it's that it just led to extra work, not an incorrect result. But we can avoid the extra work, so we should.